EpikCord/client/http_client.py

```diff
@@ -0,0 +1,193 @@
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+        """
+        super().__init__(internal_hash=internal_hash, external_hash=external_hash)
+        self.major_parameters: Dict[str, Any] = major_parameters
+
+    def __eq__(self, other: TopLevelBucket):
+        if isinstance(other, TopLevelBucket):
+            return self.major_parameters == other.major_parameters
+        return False
+
+class HTTPClient:
+    def __init__(self, token: str, *, version: int = 10):
+        self.token: str = token
+        self.version: int = version
+        self.session: aiohttp.ClientSession = aiohttp.ClientSession(headers={
+            "Authorization": f"Bot {self.token}"
+        })
+        self.buckets: Dict[str, Bucket] = {}
+        self.global_event: asyncio.Event = asyncio.Event()
+        self.global_event.set()
+
+    async def ws_connect(self, url: str) -> aiohttp.ClientWebSocketResponse:
+        return await self.session.ws_connect(url)
+
+    async def request(self, method: str, url: str, *args,  discord: bool = True, channel_id: int = 0, guild_id: int = 0, webhook_id: int = 0, webhook_token: str = "None", **kwargs) -> Optional[aiohttp.ClientResponse]:
+        """
+        Parameters
+        ----------
+        method : str
+            The method of the request.
+        url : str
+            The url of the request.
+        *args
+            The args to pass to the ClientSession.request method.
+        discord : bool
+            Whether this request is to Discord.
+        channel_id : int
+            The channel id of the request.
+        guild_id : int
+            The guild id of the request.
+        webhook_id : int
+            The webhook id of the request.
+        webhook_token : str
+            The webhook token of the request.
+        **kwargs
+            The kwargs to pass to the ClientSession.request method.
+
+        Returns
+        -------
+        Optional[aiohttp.ClientResponse]
+            The response of the request. Can be None if the request fails.
+        
+        Raises
+        ------
+        NotFound
+            If the request returns a 404 status code.
+        Forbidden
+            If the request returns a 403 status code.
+        Unauthorized
+            If the request returns a 401 status code.
+        BadRequest
+            If the request returns a 400 status code.
+        HTTPException
+            If the request returns a status code that is not OK and not any of the other exceptions.
+        """
+        if not discord:
+            return await self.session.request(method, url, *args, **kwargs)
+        if url.startswith("/"):
+            url = f"https://discord.com/api/v{self.version}{url}"
+        else:
+            url = f"https://discord.com/api/v{self.version}/{url}"
+
+        if url.endswith("/"):
+            url = url[:-1]
+        
+        bucket = self.buckets.get(url) or MockBucket()
```

Comment:
I believe it makes more sense to make a clear check here as `or` makes it looks not important